### ISSUE TITLE
Add AdaBelief optimizer [https://arxiv.org/abs/2010.07468]

### ISSUE THREAD

**Issue by TimbusCalin** [tensorflow/addons#2203]
**Describe the feature and the current behavior/state.**

**Relevant information**
- Are you willing to contribute it (yes/no): no
- Are you willing to maintain it going forward? (yes/no): no 
- Is there a relevant academic paper?  https://arxiv.org/abs/2010.07468
- Is there already an implementation in another framework? https://github.com/juntang-zhuang/Adabelief-Optimizer
- Was it part of tf.contrib? No

**Which API type would this fall under (layer, metric, optimizer, etc.): Optimizer/Layer

**Who will benefit with this feature?** Everybody; shows better results/more promising ones than the other optimizers

**Any other info.**

**Comment by AakashKumarNain** [comment]
Although `Adabelief` looks interesting but we will wait for some time before adding it here. Why? Many optimizers showed improvements over Adam in the past as well but they fail to deliver that performance when tested on wide variety of datasets. Once we get more insights about the performance, we will start the process of adding it to addons with a PR

**Comment by TimbusCalin** [comment]
Yes, I agree with your statement.

**Comment by bhack** [comment]
Yes see a recent work like https://arxiv.org/abs/2007.01547

**Comment by bhack** [comment]
TF reference implementation at https://github.com/juntang-zhuang/Adabelief-Optimizer/blob/master/TensorFlow_Experiments/AdaBelief_tf.py

**Comment by dathudeptrai** [comment]
> TF reference implementation at https://github.com/juntang-zhuang/Adabelief-Optimizer/blob/master/TensorFlow_Experiments/AdaBelief_tf.py

Seems that implementation missed many features, hope this optimizer included in tf_addons soon :D. 
```
The adabelief-tf is a naive implementation in Tensorflow. It lacks many features such as decoupled weight decay, and is not extensively tested. Currently I don't have plans to improve it since I seldom use Tensorflow, please contact me if you want to collaborate and improve it.
```

**Comment by juntang-zhuang** [comment]
> > TF reference implementation at https://github.com/juntang-zhuang/Adabelief-Optimizer/blob/master/TensorFlow_Experiments/AdaBelief_tf.py
> 
> Seems that implementation missed many features, hope this optimizer included in tf_addons soon :D.
> 
> ```
> The adabelief-tf is a naive implementation in Tensorflow. It lacks many features such as decoupled weight decay, and is not extensively tested. Currently I don't have plans to improve it since I seldom use Tensorflow, please contact me if you want to collaborate and improve it.
> ```

Hi, we have updated ```adabelief-tf==0.1.0```, which is also available by pip, and supports decoupled weight decay and rectification, and supports Tensorflow>=2.0 and Keras.

**Comment by bhack** [comment]
This Is the first mention of AdaBelief with code:
https://github.com/yuanwei2019/EAdam-optimizer

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
